LABS.md

```diff
@@ -0,0 +1,57 @@
+# LoopBack Labs
+
+`Labs` is an incubation process to build experimental features and make it easy
+for our developers to try out via `npm install`. Such features are usually not
+production ready. The design, implementation and APIs may change significantly
+over versions. We may graduate or abandon experimental features based on the
+community feedback.
+
+## Workflow
+
+We use a convention based git branching strategy to work on experimental
+features along with regular ones in `loopback-next` repository.
+
+![loopback-labs](./labs/labs.png)
+
+### Work on an experimental feature
+
+1. Set up local git repository for an experiment feature
+
+```sh
+cd loopback-next
+git fetch --all
+git checkout labs
+git checkout -b labs/<my-experimental-feature>
+git rebase origin/master
+git checkout -b labs-dev/<my-experimental-feature>
+```
+
+**NOTE**: We will create a shell script to provision an experimental feature.
+
+2. Work on an experimental feature
+
+```sh
+cd loopback-next
+git checkout labs-dev/<my-experimental-feature>
+```
+
+You can now start to make changes, commit them, and push to remote.
+
+3. Rebase the experimental feature branch against master
+
+```
+cd loopback-next
+git checkout labs-dev/<my-experimental-feature>
```

Comment:
In the experimental phase, PRs will be created from `labs-dev/<my-experimental-feature>` to `labs/<my-experimental-feature>`, does it mean people need to rebase `git rebase origin/master` for both `labs/<my-experimental-feature>` and `labs-dev/<my-experimental-feature>` all the time?

Reply:
Yes and no.

Yes - it's good to stay sync with `master` for both labs integration/dev branches
No - we can always rebase during graduation

In fact, only the `dev` branch need to rebase against master. When it merges into the integration branch for itself, the `integration` branch will be updated.

Reply:
@raymondfeng My concern is after rebasing against master, the commit history will be messed up like https://github.com/strongloop/loopback-labs/pull/6

Reply:
@raymondfeng you are right I think I am over concerned for the rebasing stuff. 

Since each feature will be in an isolated package and implemented+released on its own branch, it should just have the released production ready modules `@loopback/*` as its dependencies. And only rebase in the graduation PR.

Also, when we do code change in lab packages, tests in production ready packages shouldn't run.